eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Address/CompactAddr.hs

```diff
@@ -30,8 +36,9 @@ import Test.QuickCheck.Gen (chooseWord64)
 
 propValidateNewDecompact :: forall crypto. CC.Crypto crypto => Addr crypto -> Property
 propValidateNewDecompact addr =
-  let compact = SBS.toShort $ serialiseAddr addr
-      decompactedOld = CA.deserializeShortAddr @crypto compact
```

Comment:
Can you explain the change here? This seems like we're comparing to a different method than before. I am generally worried that we have so many different deserialisers now, it's very difficult to know which to use.

Reply:
Very good question and a very good observation: there are indeed too many deserializers now. In fact all this recent work paves the way for us remove all the old deserializers, but only after we are past the hardfork.

With regards to what has changed here is `deserialiseAddr` and `deserializeShortAddr` are almost identical, except latter works on `ShortByteString`s, while the former works on `ByteString`s. So, considering that `deserializeShortAddr` was used only for this test, we can just get rid of that function now and replace it with semantically equivalent one